### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by marcusdaly**
Changes include:
1. Include docstring-related requirements for added flake8 checks
2. Add docstrings to modules, classes, functions that were missing them
3. Adjust existing docstrings to conform to PEP257 and RST standards

### CODE REVIEW

File: fero/analysis.py [hunk 1]
```diff
@@ -170,10 +213,11 @@ def complete(self):
         return self._complete
 
     def get_results(self, format="dataframe") -> Union[pd.DataFrame, List[dict]]:
-        """Serializes the prediction results of the prediction, by default this will be a pandas
+        """Serialize the prediction results of the prediction.
```
Comment (akucukelbir) [inline fero/analysis.py:216]: Avoid the word 'serialize'. Prefer 'Return'?

File: fero/analysis.py [hunk 2]
```diff
@@ -215,56 +259,63 @@ class Analysis(FeroObject):
...
 
         return self._target_names
 
     @property
-    def _presentation_data(self):
-        """This is big and ugly, so keep it private but cached"""
+    def _presentation_data(self) -> Optional[dict]:
+        """Get the presentation data of this analysis.
+
+        This is big and ugly, so keep it private but cached.
```
Comment (akucukelbir) [inline fero/analysis.py:309]: Remove this line.

File: fero/analysis.py [hunk 3]
```diff
@@ -287,10 +338,9 @@ def _make_col_name(col_name: str, cols: List[str]) -> str:
         return col_name
 
     def is_retraining(self) -> bool:
-        """Checks if an analysis is currently be retrained.
+        """Check if an analysis is currently be retrained.
```
Comment (akucukelbir) [inline fero/analysis.py:341]: Grammar.

File: fero/analysis.py [hunk 4]
```diff
@@ -439,8 +489,9 @@ def make_prediction(
     def make_prediction_serial(
         self, prediction_data: Union[pd.DataFrame, List[dict]]
     ) -> Union[pd.DataFrame, List[dict]]:
-        """Makes a prediction from the provided data using the most recent trained model for the analysis. This
-        computes predictions one row at a time. Therefore, while it is slower than `make_prediction`, this method
+        """Make a prediction from the provided data using the most recent trained model for the analysis.
```
Comment (akucukelbir) [inline fero/analysis.py:492]: I think we no longer need this, since everyone is on v2 models now.

File: fero/asset.py [hunk 5]
```diff
@@ -77,19 +87,24 @@ class Asset(FeroObject):
...
         self._presentation_data_cache = self._client.get(
             f"/api/configuration_models/{self.latest_trained_configuration_model}/presentation_data/"
         )["data"]
 
     @property
     def _presentation_data(self):
-        """This is big and ugly, so keep it private but cached"""
+        """Get the presentation data of this asset.
+
+        This is big and ugly, so keep it private but cached
```
Comment (akucukelbir) [inline fero/asset.py:106]: Remove.

**Review by akucukelbir — CHANGES_REQUESTED** [review]
Some minor changes. Perhaps also worth merging master in?

File: fero/analysis.py (hunk 4 above)
Comment (marcusdaly) [reply fero/analysis.py]: Got it! removed this function, tests of it, and references to it in other comments.

**Comment by marcusdaly** [comment]
> Some minor changes. Perhaps also worth merging master in?

Great! Merged master in, made the changes in the docstrings, and removed `make_prediction_serial` along with any references to it. Should be good for a re-review!

**Review by akucukelbir — APPROVED** [review]

**Review by wbrinkerhoff — APPROVED** [review]

**Pull request merged**